truffle/com.oracle.truffle.api.interop/src/com/oracle/truffle/api/interop/ForeignAccess.java

```diff
@@ -87,15 +88,328 @@ public static ForeignAccess create(Factory factory) {
+        try {
+            return fn.executeForeignImpl(frame, receiver, new Object[]{identifier, value});
+        } catch (UnknownIdentifierException | UnsupportedTypeException | UnsupportedMessageException e) {
+            throw e;
+        } catch (InteropException e) {
+            throw new AssertionError("Unexpected exception catched.", e);
+        }
+    }
+
+    /**
+     * Sends a UNBOX {@link Message} to the foreign receiver object by executing the
+     * <code> unboxNode </code>.
+     *
+     * @param unboxNode the createNode created by {@link Message#createNode()}
+     * @param frame the call frame
+     * @param receiver foreign object to receive the message passed to {@link Message#createNode()}
+     *            method
+     * @return return value, if any
+     * @throws ClassCastException if the createNode has not been created by
+     *             {@link Message#createNode()} method.
+     * @throws UnsupportedMessageException if the <code>receiver</code> does not support the
+     *             {@link Message#createNode() message represented} by <code>unboxNode</code>
+     */
+    public static Object sendUnbox(Node unboxNode, VirtualFrame frame, TruffleObject receiver) throws UnsupportedMessageException {
+        ForeignObjectAccessHeadNode fn = (ForeignObjectAccessHeadNode) unboxNode;
+        try {
+            return fn.executeForeignImpl(frame, receiver);
+        } catch (UnsupportedMessageException e) {
+            throw e;
+        } catch (InteropException e) {
+            throw new AssertionError("Unexpected exception catched.", e);
+        }
+    }
+
+    /**
+     * Sends an EXECUTE {@link Message} to the foreign receiver object by executing the
+     * <code> executeNode </code>.
+     *
+     * @param executeNode the createNode created by {@link Message#createNode()}
+     * @param frame the call frame
+     * @param receiver foreign function object to receive the message passed to
+     *            {@link Message#createNode()} method
+     * @param arguments arguments passed to the foreign function
+     * @return return value, if any
+     * @throws ClassCastException if the createNode has not been created by
+     *             {@link Message#createNode()} method.
+     * @throws UnsupportedTypeException if one of element of the <code>arguments</code> has an
+     *             unsupported type
+     * @throws ArityException if the <code>arguments</code> array does not contain the right number
+     *             of arguments for the foreign function
+     * @throws UnsupportedMessageException if the <code>receiver</code> does not support the
+     *             {@link Message#createNode() message represented} by <code>executeNode</code>
+     */
+    public static Object sendExecute(Node executeNode, VirtualFrame frame, TruffleObject receiver, Object[] arguments) throws UnsupportedTypeException, ArityException, UnsupportedMessageException {
```

Comment:
Using varargs for the arguments argument might be more comfortable for the caller.